### ISSUE TITLE
Prefixes - how set in stone are they?

### ISSUE THREAD

**Issue by IsraelLand** [IAHLT/UD_Hebrew#34]
@amir-zeldes

`לבנזילפניצילין, אב הטיפוס של הפניצילינים, פעילות נגד מרבית החיידקים הגראם-חיוביים`

From what I've gathered, prefixes are only to be applied if they're of a restricted list -
`בין, תת, דו, חד, תלת, פרו, קדם, אנטי, סופר, נאו, מולטי, מיני, אקס, כלל, פוסט, טרום `

That being said, the sentence above has a very clear affix-prefix structure.
In any case, it's very similar to "complex" adjectival phrases like `מלא משמעות`, which are tagged as compounds (just backwards).
So if we assume it to be a compound anyway, the questions that remain are -

1. should we use subrel:affix for nonsanctioned words/prefixes like "gram"?
2. should we apply prefix=yes?
3. should we tag "gram" as ADV, then? Otherwise it'd be a noun, which I find not as accurate, though I'm no M.D.

Thank you!

**Comment by amir-zeldes** [comment]
I'm not opposed to more prefixes, but since the closed list has served us so well in being exhaustive, we may want to document this and continue treating it as a closed list based category.

**Comment by IsraelLand** [comment]
Great, but the main thing is, should we just add any prefix we encounter to the list? 
Especially since "gram" is not by any means a common prefix, but it still acts like one. 
This effectively means a long list down the line, which kinda takes the point out of or changes the direction of the exclusive list we have now.
Perhaps a sub-list with more allowed prefixes of more obscure cases (I mean, people down the line would never understand what the prefix "gram-" means out of context)?
Thank you

**Comment by amir-zeldes** [comment]
I see what you mean, but although rare items are added, additions are themselves rare, so I think maintaining the list is worth it. I don't mind if you split it into sub-groups (common/rare)

**Comment by IsraelLand** [comment]
Great, that'll work. Thanks!

**Comment by Hilla-Merhav** [comment]
@amir-zeldes 
Can we also add כל/כול to the prefix list? for cases like - כל-יכול, כל-יודע? 
And היפר for היפראקטיבי, היפר ונטילציה?

**Comment by amir-zeldes** [comment]
That seems reasonable. Though if they start to be common, and are of course always prefixes, I'm starting to wonder if we actually need this subtype, since you could find all of these even if just labeled `compound` by looking for a compound dominated from second token to first (instead of the usual smixut direction)

**Comment by Hilla-Merhav** [comment]
Hi @amir-zeldes

I have a question about prefixes' POS – this issue is unclear in the [Morphological Features Table](https://github.com/ivrit/IAHLT-HTB-GUIDELINES/blob/main/Morphological%20features.md). 

The _prefix_ feature appears twice in the table: once below the ADV category (Modifying an ADJ: דו-קוטבי, תלת-ספרתי, רב-לאומית ADV: חד-חד-ערכי VERB: אי אפשר) and once below the ADJ category – according to the table we assign Prefix=Yes to ADJs also, and the selected examples are: אנטי-קומוניסט, מולטי-מיליונר.

I think אנטי and מולטי are also ADVs, not ADJs – these forms show no agreement in Number, Gender or Definiteness. On HTB אנטי is usually tagged **ADV** (12 occurrences) and only one time ADJ. מולטי has only 2 occurrences (ADJ) - I believe we should fix them. I don't see any differences between רב and מולטי, and I believe it's better these forms would share one POS. 

Also, I am adding information to the [Morphological Features Table](https://github.com/ivrit/IAHLT-HTB-GUIDELINES/blob/main/Morphological%20features.md) to let Nick write some level-4 rules for the ud-tools validate script, and I want to say Number and Gender are mandatory features of ADJs. If we tag מולטי and אנטי as ADJ, I think (we'll test it, of course) these the only cases that prevent me from saying it. 

Is it OK I add מולטי to the ["amending the HTB" list](https://docs.google.com/document/d/1_GaUIwt0kiSh3AwE8B47m7ARNF-PHPxC/edit), and remove the prefix feature from the ADJ category in the [Morphological Features Table](https://github.com/ivrit/IAHLT-HTB-GUIDELINES/blob/main/Morphological%20features.md)?

**Comment by amir-zeldes** [comment]
That's a good question, and from a strictly theoretical point of view I think there are strong arguments that say these things have no real POS. The notion of POS is heavily related to headedness, since virtually any POS test we can imagine requires some context in which that thing is the head. For compound modifiers, if the modifier can never appear as a head, it is questionable whether it even has a POS, or is it really just a morphological process attaching it to the stem before we can even speak of POS categories.

This is not necessarily true of head affixes: consider by contrast something like English -er in "driver". You could say "drive" is a verb, "-er" is a nominal suffix, and that is why their combination results in a noun, so "-er", if we had to tag it, is a noun. But what about non-head prefixes like, say, "un"? By convention, some people assume because "un-available" is an adjective, "un" must be an adjective prefix too, but really there is no way to tell, since POS tests would only apply to the entire thing.

I think it's a similar logic which leads to "anti-" -> ADJ, but "i-(efshar)" -> ADV. We imagine the an adjective in anti- is compatible with anti- because it too is an ADJ, and maybe we identify "i-" with "lo", so as a negation it is ADV too. 

Given the need to assign gender and number to adjectives, I think ADV for all of these is not a bad solution (if also somewhat arbitrary), but I'd like to hear if there are dissenting voices as well. If so, we would have to say there is a class of "sub-word-level" ADJ prefixes, which are gender and number less but still tagged ADJ.

**Comment by IsraelLand** [comment]
I've always found the "ADJ before noun, ADV before adj." system a bit arbitrary, and I definitely agree that those are in no way real adjectives.
That is not to say tagging them ADV from now on isn't arbitrary, but I think your solution where we keep this system, on top of always tagging non-agreeing, genderless, POS-less _(these should be the tests, I guess)_ prefixes as ADV, is much better.

**Comment by amir-zeldes** [comment]
I'm OK with it as long as HTB gets revised - we're starting to amass a lot of promises for revisions, so I'm getting a little worried whether it will be realistic to apply all of these decisions to HTB! (even though some of them are not particularly difficult)

**Comment by Hilla-Merhav** [comment]
@amir-zeldes Technically, is it possible we change every token that is tagged ADJ and its deprel is `compound:affix` to ADV? If we can do it at once, it might not be such a commitment.

**Comment by amir-zeldes** [comment]
Yes, it sounds like that is the most popular plan. I'm OK with it. Could you point me to where the list of accepted planned HTB edits is? I may be able to run some of these easily on the version in GitHub, unless someone has already been editing it concurrently. @ivrit - Has there been some work making the github IAHLT HTB out of date?

**Comment by Hilla-Merhav** [comment]
@amir-zeldes Sure, [here ](https://docs.google.com/document/d/1_GaUIwt0kiSh3AwE8B47m7ARNF-PHPxC/edit) is the list, and as far as I know the HTB hasn't been edited yet :)

**Comment by ivrit** [comment]
@amir-zeldes no, no work on it as yet :)

**Comment by IsraelLand** [comment]
@amir-zeldes should `רב, חד` and the like be seperated from these, tagged in "the old way" of adv/adj for adj/noun respect., since they are usually adjectival and not POS-less (the kind of adjectives not requiring -i), or would that be too much of a trouble to make a distinction?

**Comment by Hilla-Merhav** [comment]
@israel רב is not always a prefix (I think it rarely comes as a prefix).
When רב is in a structure of exocentric compound adjectives of the bahuvrīhi type (e.g. רב-ערך, כחולי-עיניים, ארוכת-צוואר) – the whole unit in a position of `amod`, so the ADJ רב must be a head:
מידע רב ערך
`amod`(meyda, rav)
`compound`(rav, erex)
The only example of a prefix רב I found until now is רב-סוכר –I believe its origin is also the structure of exocentric compound adjective, but the whole unit רב-סוכר is a NOUN and not an ADJ, so I believe סוכר should be the head. 
We can simply say that when רב is a head it's an ADJ (and then Gender and Number are relevent - מידע רב ערך, מחשבה רבת ערך), and when it's a `compound:affix` it's an ADV.

**Comment by IsraelLand** [comment]
We can make the distinction that they are not the same (like the same beinoni is sometimes verb and sometimes isn't), depends if we decide to. My point is mainly akin to the fact that we tag "the correct POS" even for fixed expressions, even when the head is the first word and we don't **really** care about the POS of the ones inside. Question is, is רב the same as רב?

**Comment by amir-zeldes** [comment]
I think they are not the same one in these examples (rav-sukar != rav erex), and there is no problem tagging them based on context (one is a prefix, so "ADV" by the latest guideline, and the other is an ADJ in a compound). Notice also semantically, rav-sukar is a type of sukar, but rav-erex is not a type of erex.

**Comment by IsraelLand** [comment]
Great, so we basically abandon the old system in favor of always tagging ADV (can't think of other cases). Thanks for the clarification.

**Comment by Hilla-Merhav** [comment]
Hi @amir-zeldes

Testing this validation rule: **only ADJ can be governed by `amod`**, the query retrieves this kind of utterances:

דרום קוריאני
בית ספרי
דרום מזרחי
תל אביבי
מזרח תיכוני

The utterance מאכל מזרח תיכוני, for example, is analyzed this way:
![מאכל מזרח תיכוני](https://user-images.githubusercontent.com/90029994/149388626-d6fac29e-9f88-47e8-9ca9-707a5e8ea1ca.JPG)

So the noun in these trees takes the `amod` deprel. 
Do you think these adjectives should be analyzed right-headed at all?
(BTW in most of these utterances the original structure is indeed smixut, but מזרח תיכוני comes from מזרח תיכון – an `amod` relation, so I suggest different tree for more than one reason.)

I suggest a left-headed analysis: 
`amod`(maaxal, tixoni)
`compound:affix`(tixoni, mizrax)
Or maybe just `compound`(tixoni, mizrax) –  this depends on whether we really need this subtype - I am raising this here because this very question has already been raised in this discussion:

>if they start to be common, and are of course always prefixes, I'm starting to wonder if we actually need this subtype, since you could find all of these even if just labeled compound by looking for a compound dominated from second token to first (instead of the usual smixut direction)

**Comment by amir-zeldes** [comment]
I agree with your analysis - this is a compound adjective, headed by "tixoni".

**Comment by Hilla-Merhav** [comment]
@amir-zeldes Thanks! Do you think the relation should be `compound` or `compound:affix`?
So far, we use `compound:affix` for left-headed compound structures, and `compound` for right-headed. We treat `compound:affix` dependents as a closed list based category – בין, תת, דו, חד, תלת, פרו, קדם, אנטי, סופר etc, and assign to them the feature Prefix=Yes (all of them are tagged ADV).

Given that adjectives like מזרח-תיכוני belong to an open class, I think we have three options:
1.	We can distinguish between the closed list of prefixes (פרו, קדם, אנטי) and structures like בית-ספרי, תל-אביבי, דרום-אמריקאי, מזרח-תיכוני. The prefixes would still be governed through `compound:affix` and tagged ADV with `Prefix=Yes`. As for the left-headed adjectives, like mizrax-tixoni – mizrax would be governed through `compound` (and be tagged NOUN). 
2.	We can remove the `compound:affix` subrel, and use the `compound` relation for left-headed compound structures in general. We can still use `Prefix=Yes` with the closed list of prefixes (פרו, קדם, אנטי), not with בית, תל, דרום, מזרח; that means we still have a separated list of prefixes.
3.	We can remove the `compound:affix` subrel and also `Prefix=Yes` feature, and use the `compound` relation for left-headed compound structures in general. We can treat prefixes as an open class.


As for the left-headed adjectives like דרום-אמריקאי, I think we should anyway tag מזרח, דרום, בית etc NOUN. If their origin is smixut (בית-ספרי, דרום-אמריקאי) I think we should assign `Definite=Cons` as usual to the dependents (like beit in beit-sifri). What do you say?

When it comes to PROPN-origin adjectives, like `Tel-Aviv` (PROPN+PROPN) –> `Tel-Avivi`; The head, `Avivi`, is tagged ADJ, but do you think `Tel` should stay PROPN?

**Comment by amir-zeldes** [comment]
My preference would be:

  * Use regular `compound` for mizrax-tixoni, since it's not an affix
  * Use Definite=Cons for those cases that are underlyingly smixut (researchers interested in smixut should arguably be able to find these together, since they instantiate smixut)
  * Tel should be PROPN, since I interpret the universal guideline to say "if it's a name and you would have otherwise tagged it NOUN, then it is PROPN"
